test/integration/command/source_test.py

```diff
@@ -142,6 +142,30 @@ def source(self):
         self.assertIn("conanfile.py: Running source!", client.out)
         self.assertEqual("Hello World", client.load("file1.txt"))
 
+    def test_local_source_layout_root(self):
+        # Ensure that if a root folder is specified Conan source method
+        # uses it. This is especially important when running with
+        # no_copy_source where both the build()  and source() methods
+        # need to reference source_folder (so the value must be the
+        # same in both methods)
+        conanfile = textwrap.dedent('''
+            from conan import ConanFile
+
+            class ConanLib(ConanFile):
+                
+                def layout(self):
+                    self.folders.root = "root_folder"
+
+                def source(self):
+                    self.output.info(f"Source Folder: {self.source_folder}")
+        ''')
+        # First, failing source()
+        client = TestClient(light=True)
+        client.save({CONANFILE: conanfile})
+
+        client.run("source .")
+        self.assertTrue(client.out.strip().endswith("spaces\\root_folder"))
```

Comment:
This is most likely broken in other systems such as Linux, it seems you developed and tested it under Windows?
In any case, lets go back to the issue, to clarify the goal and the intent, and then we will go back to the PR code, thanks!